pylint/checkers/variables.py

```diff
@@ -1062,6 +1070,30 @@ class VariablesChecker(BaseChecker):
                 "help": "List of names allowed to shadow builtins",
             },
         ),
+        (
+            "timeout-methods",
+            {
+                "default": (
+                    "http.client.HTTPConnection",
+                    "http.client.HTTPSConnection",
+                    "requests.delete",
+                    "requests.get",
+                    "requests.head",
+                    "requests.options",
+                    "requests.patch",
+                    "requests.post",
+                    "requests.put",
+                    "requests.request",
+                    "serial.Serial",
+                    "smtplib.SMTP",
+                    "urllib.request.urlopen",
```

Comment:
Let's remove this. 
https://docs.python.org/3/library/urllib.request.html#urllib.request.urlopen
The documentation states that the default timeout will be used if it is not specified. So there is no risk of hanging I think. Or at least, we have no way to determine if the global timeout is indeed set so we shouldn't warn about it.

Edit: I only just saw that you already identified this issue. I agree, let's remove them.